wren-ai-service/src/pipelines/common.py

```diff
@@ -506,3 +745,110 @@ def dry_run_pipeline(pipeline_cls: BasicPipeline, pipeline_name: str, **kwargs):
+        type: Literal["bar", "line", "area", "arc"]
+
+    class ChartData(BaseModel):
+        values: list[dict]
+
+    class ChartEncoding(BaseModel):
+        field: str
+        type: Literal["ordinal", "quantitative", "nominal"]
+        title: str
+
+    schema: str = Field(
+        alias="$schema", default="https://vega.github.io/schema/vega-lite/v5.json"
+    )
+    title: str
+    data: ChartData
+    mark: ChartType
+    encoding: ChartEncoding
+
+
+class TemporalChartEncoding(ChartSchema.ChartEncoding):
+    type: Literal["temporal"] = Field(default="temporal")
+    timeUnit: str = Field(default="yearmonth")
+
+
+class LineChartSchema(ChartSchema):
+    class LineChartMark(BaseModel):
+        type: Literal["line"] = Field(default="line")
+
+    class LineChartEncoding(BaseModel):
+        x: TemporalChartEncoding | ChartSchema.ChartEncoding
+        y: ChartSchema.ChartEncoding
+        color: ChartSchema.ChartEncoding
+
+    mark: LineChartMark
+    encoding: LineChartEncoding
+
+
+class BarChartSchema(ChartSchema):
+    class BarChartMark(BaseModel):
+        type: Literal["bar"] = Field(default="bar")
+
+    class BarChartEncoding(BaseModel):
+        x: TemporalChartEncoding | ChartSchema.ChartEncoding
+        y: ChartSchema.ChartEncoding
+        color: ChartSchema.ChartEncoding
+
+    mark: BarChartMark
+    encoding: BarChartEncoding
+
+
+class GroupedBarChartSchema(ChartSchema):
+    class GroupedBarChartMark(BaseModel):
+        type: Literal["bar"] = Field(default="bar")
+
+    class GroupedBarChartEncoding(BaseModel):
+        x: TemporalChartEncoding | ChartSchema.ChartEncoding
+        y: ChartSchema.ChartEncoding
+        xOffset: ChartSchema.ChartEncoding
+        color: ChartSchema.ChartEncoding
+
+    mark: GroupedBarChartMark
+    encoding: GroupedBarChartEncoding
+
+
+class StackedBarChartYEncoding(ChartSchema.ChartEncoding):
+    stack: Literal["zero"] = Field(default="zero")
+
+
+class StackedBarChartSchema(ChartSchema):
+    class StackedBarChartMark(BaseModel):
+        type: Literal["bar"] = Field(default="bar")
+
+    class StackedBarChartEncoding(BaseModel):
+        x: TemporalChartEncoding | ChartSchema.ChartEncoding
+        y: StackedBarChartYEncoding
+        color: ChartSchema.ChartEncoding
+
+    mark: StackedBarChartMark
+    encoding: StackedBarChartEncoding
+
+
+class PieChartSchema(ChartSchema):
+    class PieChartMark(BaseModel):
+        type: Literal["arc"] = Field(default="arc")
+
+    class PieChartEncoding(BaseModel):
+        theta: ChartSchema.ChartEncoding
+        color: ChartSchema.ChartEncoding
+
+    mark: PieChartMark
+    encoding: PieChartEncoding
+
+
+class AreaChartSchema(ChartSchema):
+    class AreaChartMark(BaseModel):
+        type: Literal["area"] = Field(default="area")
+
+    class AreaChartEncoding(BaseModel):
+        x: TemporalChartEncoding | ChartSchema.ChartEncoding
+        y: ChartSchema.ChartEncoding
+
+    mark: AreaChartMark
+    encoding: AreaChartEncoding
```

Comment:
im considering about moving these chart schema class to an independent module. wdyt?

Reply:
fixed